docs/book/src/topics/bring-your-own-aws-infrastructure.md

```diff
@@ -274,3 +274,67 @@ The external system must provide all required fields within the spec of the AWSC
 Once the user has created externally managed AWSCluster, it is not allowed to convert it to CAPA managed cluster. However, converting from managed to externally managed is allowed.
 
 User should only use this feature if their cluster infrastructure lifecycle management has constraints that the reference implementation does not support. See [user stories](https://github.com/kubernetes-sigs/cluster-api/blob/10d89ceca938e4d3d94a1d1c2b60515bcdf39829/docs/proposals/20210203-externally-managed-cluster-infrastructure.md#user-stories) for more details.
+
+
+## Bring your own (BYO) Public IPv4 addresses
+
+Cluster API also provides a mechanism to allocate Elastic IP from the existing Public IPv4 Pool that you brought to AWS[1].
+
+Bringing your own Public IPv4 Pool (BYOIPv4) can be used as an alternative to IPv4 charges introduced by AWS in February 2024[2].
+
+Supported resources to BYO Public IPv4 Pool (`BYO Public IPv4`):
+- Nat Gateways
+- Network Load Balancer for API server
+- Machines
+
+Use `BYO Public IPv4` when you have brought to AWS custom IPv4 CIDR blocks and want to cluster automatically from custom pools instead of Amazon-provided pools.
```

Comment:
```suggestion
Use `BYO Public IPv4` when you have brought to AWS custom IPv4 CIDR blocks and want the cluster to automatically use IPs from the custom pool instead of Amazon-provided pools.
```